src/clojure/nrepl/middleware/interruptible_eval.clj

```diff
@@ -63,13 +64,22 @@
     (if (and ns (not explicit-ns))
       (t/send transport (response-for msg {:status #{:error :namespace-not-found :done}
                                            :ns ns}))
-      (let [ctxcl (.getContextClassLoader (Thread/currentThread))]
+      (let [ctxcl (.getContextClassLoader (Thread/currentThread))
+            ^Writer out (->PrintWriter :out msg)
+            ^Writer err (->PrintWriter :err msg)]
         (try
           (clojure.main/repl
            :eval (if eval (find-var (symbol eval)) clojure.core/eval)
            :init #(let [bindings
                         (-> (get-thread-bindings)
                             (into @session)
+                            (into {#'*out* out
+                                   #'*err* err
+                                   ;; clojure.test captures *out* at load-time, so we need to make sure
+                                   ;; runtime output of test status/results is redirected properly
+                                   ;; TODO: is this something we need to consider in general, or is this
+                                   ;; specific hack reasonable?
+                                   #'clojure.test/*test-out* out})
```

Comment:
This comment was copied from `create-session` where the bindings were established previously – so the behaviour has been in nREPL for several years. I think it's the right thing to do. `clojure.test` is part of Clojure and it's common to run tests at the REPL. It would be otherwise up to the user to somehow make sure `clojure.test` captures the right value of `*out*` at load-time.

Reply:
I guess we can remove the TODO then. :-)